api/src/controllers/changes.js

```diff
@@ -189,13 +191,15 @@ const getChanges = feed => {
     }
   }
 
-  feed.changesReqs = [];
+  const upstreamSession = { id:upstreamSessionId, requests:[] };
+  feed.upstreamSessions[upstreamSessionId] = upstreamSession;
```

Comment:
When would you have more than one upstream session? I think the second one should cancel the first one so there's only ever one upstream request for each chunk for each feed, rather than multiple per chunk.